src/taipy/gui_core/viselements.json

```diff
@@ -55,6 +55,12 @@
             "type": "str",
             "default_value": "50vh",
             "doc": "The maximum height, in CSS units, of the tree."
+          },
+          {
+            "name": "on_creation",
+            "type": "Callback",
+            "doc": "The name of the function that is triggered when a scenario will be or has been created.<br/><br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (str): the identifier of the button.</li>\n<li>action (str): the name of the action that provoked the change.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>before: if True, the action is invoked before the creation and the payload contains the parameters entered by the user (config, date, name, properties).</li>\n<li>after: if True, the action is invoked after the creation and the payload contains the scenario.</li>\n</ul>\n</li>\n<li>return: if before is True, the callback can return a scenario, an error message (the scenario will not be craeted) or None (the scenario will be created with the user parameters).</li>\n</ul>",
```

Comment:
You're talking about the button id? What button? What id?

No more before/after in the payload if I'm correct?

More globally, I guess this will be both simpler, and more complicated hu?